Recruit-Training/Embedded-Recruit-Training/Core/Src/GreenLedToggleTask.c

```diff
@@ -7,11 +7,39 @@ void greenLedToggleTask(void const* arg)
 
     osMutexId_t* canMutex = (osMutexId_t*)arg;
 
+    static const uint32_t GREEN_LED_STATUS_STDID = 0xDDD;
+    static const uint32_t GREEN_LED_TOGGLE_FREQ = 100;
+
     for (;;)
     {
         //TODO: Add GREEN_LED_TOGGLE_FREQ to prevWakeTime
+        prevWakeTime += GREEN_LED_TOGGLE_FREQ;
         osDelayUntil(prevWakeTime);
-        //TODO: Check blue toggel flag and toggle blue LED
+
+        //TODO: Check green toggle flag and toggle green LED
+        if(greenLedToggleFlag) 
+        {
+            HAL_GPIO_TogglePin(GPIOA, LED_GREEN_Pin);
```

Comment:
`GPIOA` is the value but you want to use the variable which is `LED_GREEN_GPIO_Port` since `GPIOA` makes less sense than `LED_GREEN_GPIO_Port`. When you `#define something blah` you set `something` to be `blah`.